forge-webhook-parser.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python3
+"""
+Parse a webhook payload file to clone the git repository to
+the current directory.
+"""
+
+import argparse
+import json
+import shutil
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional
+
+
+@dataclass
+class PayloadFields:
+    """
+    Dataclass of details extracted from the payload.
+    """
+
+    added: List[str]
+    removed: List[str]
+    modified: List[str]
+    default_repo_url: Optional[str] = None
+    default_clone_path: Optional[Path] = None
+
+
+PAYLOAD_FILE = "./github-payload.json"
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-f", "--file", help="webhook payload file to parse", default=PAYLOAD_FILE
+)
+args = parser.parse_args()
+
+with open(args.file, "r", encoding="utf-8") as file:
+    data = json.load(file)
+
+repo_url_keys = []
+
+if "repository" in data and "clone_url" in data["repository"]:
+    repo_url_keys.append(data["repository"]["clone_url"])
+
+if "project" in data and "git_http_url" in data["project"]:
+    repo_url_keys.append(data["project"]["git_http_url"])
+
+default_repo_url = next((url for url in repo_url_keys if url), "")
```

Comment:
I think I don't understand this section (starting from line 40) correctly.
Is there a case in which we can have more than one item in `repo_url_keys`? To me, it seems that you

1. Create an empty list
2. Add one item to the list
3. Take the first item from the list or an empty string

So I would say we don't need the list in the first place. But maybe I am missing something?

Reply:
haha. well, when you say it like that... it does seem pretty flaky. I don't know, I think I got a bit of tunnel vision and maybe fixated on the two json files that I've been playing around with.

I'll spend some more time on this to rework it. thanks for the feedback.

Reply:
LOL, yeah the list was completely unnecessary. I think maybe I was imagining the need to parse multiple json files at once or something. I've just pushed an update that hopefully looks a little more sane. thanks!

Reply:
whoops. I guess I should have also added an `else` clause in case. not sure if `""` or `None` is better though.

Reply:
I'd go with `None` but it's a detail